numpy/_core/src/umath/clog1p_wrappers.cpp.src

```diff
@@ -0,0 +1,32 @@
+#include <Python.h>
+#define NPY_NO_DEPRECATED_API NPY_API_VERSION
+#include "numpy/npy_math.h"
+#include "numpy/ndarraytypes.h"
+#include <cmath>
+#include <complex>
+#include "log1p_complex.h"
+
+extern "C" {
+
+/**begin repeat
+ * #c_fp_type = float, double, long double#
+ * #np_cplx_type = npy_cfloat,npy_cdouble,npy_clongdouble#
+ * #c = f, , l#
+ */
+
+/*
+ * C wrapper for log1p_complex(z).  This function is to be used only
+ * when the input is close to the unit circle centered at -1+0j.
+ */
+NPY_NO_EXPORT void
+clog1p@c@(@np_cplx_type@ *x, @np_cplx_type@ *r)
```

Comment:
Would have a mild preference to giving this a different name. `clog1p` isn't clear that it is only relevant in parts of the domain.

Might be nice to avoid the `.src` style templating,  I think that would mean making this a `static (inline)` templated function and then creating three explicit stubs that call it below.
Since each stub has a single line, I would hope it isn't significantly longer.